utils/common/src/main/java/org/apache/brooklyn/util/guava/Maybe.java

```diff
@@ -457,6 +457,8 @@ public boolean equals(Object obj) {
         Maybe<?> other = (Maybe<?>)obj;
         if (!isPresent()) 
             return !other.isPresent();
+        if (!other.isPresent())
+            return false;
```

Comment:
I suppose it's useful when you put it inside containers (i.e. `remove(Object)` works for sets)